File: data/Converters/ValueConverter/HalfWidthConverter.vb
Imports System.Globalization
Imports System
Imports System.Windows.Data

Namespace DPC.Data.Converters.ValueConverter
    Public Class HalfWidthConverter
        Implements IValueConverter

        Public Function Convert(value As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.Convert
            Dim fullWidth As Double = CType(value, Double)
            Return fullWidth / 2 ' Return half of the navbar width
        End Function

        Public Function ConvertBack(value As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
            Return DependencyProperty.UnsetValue
        End Function
    End Class
End Namespace

File: DPC.Data.Converters/HalfWidthConverter.cs
using System;
using System.Globalization;
using System.Windows.Data;

namespace DPC.DPC.Data.Converters.ValueConverter
{
    public class HalfWidthConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
        {
            if (value is double width)
            {
                return width / 2;
            }
            return 0;
        }

        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
        {
            throw new NotImplementedException();
        }
    }
}